### DISCUSSION TITLE
Handling lockfile updates from on different platform than one mentioned in the Gemfile.lock

### DISCUSSION THREAD

**Discussion by RahulGautamSingh** [ruby/rubygems#8014] [Q&A]
I have a `Gemfile.lock` which has the gem `wdm (0.1.1)` and platform as `x86_64-linux`. (wdm is an indirect dep so I cannot run `bundler update wdm`)

I am currently on a windows system and want to update the `wdm` dep.

I ran the command `bundler update --all`, which resulted in the following error:

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

   current directory: C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/wdm-0.1.1/ext/wdm 
```

What is the reason for this and how can I resolve this issue?

I also tried the following commands but didn't get an update for `wdm`, while other gems were updated.

```
bundler lock --update --add-platform x86_64-linux --remove-platform x64-mingw-ucrt
```

**Comment by deivid-rodriguez** [comment]
You should be able to update an indirect dependency like `wdm` with `bundle update wdm`. It sounds like you're having trouble actually installing the update (probably missing some native tool?). If all you want is an update to the lockfile, you can indeed use `bundle lock`. I think `bundle lock --update wdm` should do the trick?

**Comment by deivid-rodriguez** [comment]
Closing since I never got an answer.
